src/server/tasks/sync-campaign-contact-to-van.ts

```diff
@@ -0,0 +1,223 @@
+        ),
+        points as (
+          -- Result Codes
+          select
+            configured_response_values.canvassed_at,
+            json_build_object(
+              'result_code_id', external_result_code.external_id
+            )::jsonb as result_code,
+            null::jsonb as activist_code,
+            null::jsonb as response_option
+          from configured_response_values
+          join external_sync_config_question_response_result_code qrrc
+            on qrrc.question_response_config_id = configured_response_values.id
+          join external_result_code
+            on external_result_code.id = qrrc.external_result_code_id
+        
+          union
+        
+          -- Activist Codes
+          select
+            configured_response_values.canvassed_at,
+            null::jsonb as result_code,
+            json_build_object(
+              'activist_code_id', external_activist_code.external_id
+            )::jsonb as activist_code,
+            null::jsonb as response_option
+          from configured_response_values
+          join external_sync_config_question_response_activist_code qrac
+            on qrac.question_response_config_id = configured_response_values.id
+          join external_activist_code
+            on external_activist_code.id = qrac.external_activist_code_id
+        
+          union
+        
+          -- Survey Question Response Options
+          select
+            configured_response_values.canvassed_at,
+            null::jsonb as result_code,
+            null::jsonb as activist_code,
+            json_build_object(
+              'survey_question_id', external_survey_question.external_id,
+              'response_option_id', external_survey_question_response_option.external_id
+            )::jsonb as response_option
+          from configured_response_values
+          join external_sync_config_question_response_response_option qrro
+            on qrro.question_response_config_id = configured_response_values.id
+          join external_survey_question_response_option
+            on external_survey_question_response_option.id = qrro.external_response_option_id
+          join external_survey_question
+            on external_survey_question.id = external_survey_question_response_option.external_survey_question_id
+        )
+        select
+          date_trunc('day', canvassed_at) as canvassed_at,
+          array_to_json(array_remove(array_agg(result_code), null)) as result_codes,
+          array_to_json(array_remove(array_agg(activist_code), null)) as activist_codes,
+          array_to_json(array_remove(array_agg(response_option), null)) as response_options
+        from points
+        group by 1
+      `,
+      [contactId]
+    )
+    .then(({ rows }: { rows: CanvassResultRow[] }) => rows);
+
+  if (canvasResultsRaw.length === 0) return;
+
+  let chain = Promise.resolve();
+  for (const canvassResult of canvasResultsRaw) {
+    chain = chain.then(async () => {
```

Comment:
I think just putting the `await` inside the for loop is easier to read that the chain pattern.

Shouldn't we just be making one request per contact though?